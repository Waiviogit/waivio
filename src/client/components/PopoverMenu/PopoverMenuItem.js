import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './PopoverMenuItem.less';

const PopoverMenuItem = ({
  itemKey,
  children,
  onClick,
  bold,
  disabled,
  fullScreenHidden,
  topNav,
}) => (
  <li
    className={classNames('PopoverMenuItem', {
      'PopoverMenuItem--bold': bold,
      'PopOverMenuItem__full-screen-hidden': fullScreenHidden,
      PopOverMenuitem__topNav: topNav,
    })}
  >
    <a
      role="presentation"
      disabled={disabled}
      onClick={e => {
        e.preventDefault();
        onClick(itemKey);
      }}
    >
      {children}
    </a>
  </li>
);

PopoverMenuItem.propTypes = {
  children: PropTypes.node,
  itemKey: PropTypes.string,
  onClick: PropTypes.func,
  bold: PropTypes.bool,
  disabled: PropTypes.bool,
  fullScreenHidden: PropTypes.bool,
  topNav: PropTypes.bool,
};

PopoverMenuItem.defaultProps = {
  children: null,
  itemKey: '',
  onClick: () => {},
  bold: true,
  disabled: false,
  fullScreenHidden: false,
  topNav: false,
};

export default PopoverMenuItem;
