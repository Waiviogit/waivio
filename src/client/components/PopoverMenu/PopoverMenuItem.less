@import (reference) '../../styles/modules/variables.less';
@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/mixins.less';

.PopoverMenuItem {
  height: 30px !important;
  font-size: 16px !important;
  font-weight: 500 !important;

  a {
    .menu-item-link();

    &--bold {
      & > a {
        font-weight: bold !important;
      }
    }
  }
}

.PopoverMenuItem.PopOverMenuItem__full-screen-hidden {
  display: list-item;

  @media @small {
    display: none;
  }
}

.PopoverMenuItem.PopOverMenuitem__topNav {
  font-size: 18px !important;
  font-weight: 400 !important;
}
