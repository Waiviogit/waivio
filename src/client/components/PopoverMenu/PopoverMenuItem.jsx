import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './PopoverMenuItem.less';

const PopoverMenuItem = ({
  itemKey,
  children,
  onClick,
  bold,
  disabled,
  fullScreenHidden,
  topNav,
  invisible,
  active,
}) => (
  <li
    className={classNames('PopoverMenuItem', {
      'PopoverMenuItem--bold': bold,
      'PopoverMenuItem--active': active,
      'PopoverMenuItem--invisible': invisible,
      'PopOverMenuItem__full-screen-hidden': fullScreenHidden,
      PopOverMenuitem__topNav: topNav,
    })}
  >
    <a
      role="presentation"
      disabled={disabled}
      onClick={e => {
        e.preventDefault();
        onClick(itemKey);
      }}
    >
      {children}
    </a>
  </li>
);

PopoverMenuItem.propTypes = {
  children: PropTypes.node,
  itemKey: PropTypes.string,
  onClick: PropTypes.func,
  bold: PropTypes.bool,
  disabled: PropTypes.bool,
  active: PropTypes.bool,
  fullScreenHidden: PropTypes.bool,
  topNav: PropTypes.bool,
  invisible: PropTypes.bool,
};

PopoverMenuItem.defaultProps = {
  children: null,
  itemKey: '',
  onClick: () => {},
  bold: true,
  disabled: false,
  active: false,
  fullScreenHidden: false,
  topNav: false,
  invisible: false,
};

export default PopoverMenuItem;
