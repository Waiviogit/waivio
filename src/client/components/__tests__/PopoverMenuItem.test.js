import React from 'react';
import { shallow } from 'enzyme';
import PopoverMenuItem from '../PopoverMenu/PopoverMenuItem';

describe('(Component) PopoverMenuItem', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<PopoverMenuItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
