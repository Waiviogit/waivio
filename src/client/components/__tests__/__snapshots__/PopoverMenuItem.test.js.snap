// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) PopoverMenuItem renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopoverMenuItem
    bold={true}
    disabled={false}
    fullScreenHidden={false}
    itemKey=""
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        disabled={false}
        onClick={[Function]}
        role="presentation"
      />,
      "className": "PopoverMenuItem PopoverMenuItem--bold",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "disabled": false,
        "onClick": [Function],
        "role": "presentation",
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          disabled={false}
          onClick={[Function]}
          role="presentation"
        />,
        "className": "PopoverMenuItem PopoverMenuItem--bold",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "disabled": false,
          "onClick": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": null,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
